spec/controllers/api/v1/tokens_controller_spec.rb

```diff
@@ -3,6 +3,113 @@
 require 'rails_helper'
 
 RSpec.describe API::V1::TokensController, type: :controller do
+  describe 'POST#create' do
+    context 'given a valid oauth application' do
+      context 'given valid user credentials' do
+        it 'returns success status' do
+          application = Fabricate(:application)
+          user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
+
+          params = {
+            grant_type: 'password',
+            email: 'dev@nimblehq.co',
+            password: '12345678'
+          }.merge(oauth_application_params(application))
+
+          post :create, params: params
+
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns an empty response' do
+          application = Fabricate(:application)
+          user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
+
+          params = {
+            grant_type: 'password',
+            email: 'dev@nimblehq.co',
+            password: '12345678'
+          }.merge(oauth_application_params(application))
+
+          post :create, params: params
+
+          response_body = JSON.parse(response.body)
+          expect(response_body).to match_json_schema('v1/tokens/create/valid')
+        end
+      end
+
+      context 'given invalid user credentials' do
+        it 'returns bad_request status' do
+          application = Fabricate(:application)
+
+          params = {
+            grant_type: 'password',
+            email: 'dev@nimblehq.co',
+            password: '12345678'
+          }.merge(oauth_application_params(application))
+
+          post :create, params: params
+
+          expect(response).to have_http_status(:bad_request)
+        end
+
+        it 'returns an empty response' do
+          application = Fabricate(:application)
+
+          params = {
+            grant_type: 'password',
+            email: 'dev@nimblehq.co',
+            password: '12345678'
+          }.merge(oauth_application_params(application))
+
+          post :create, params: params
+
+          response_body = JSON.parse(response.body)
+          expect(response_body).to match_json_schema('v1/tokens/create/invalid')
+        end
+      end
+    end
+
+    context 'given an invalid oauth application' do
+      it 'returns unauthorized status' do
+        user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
+
+        params = {
+          grant_type: 'password',
+          email: 'dev@nimblehq.co',
+          password: '12345678'
+        }
+
+        post :create, params: params
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+
+      it 'returns an error message' do
+        user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/29430cc9172c80e61e2206f65e8842ea68b7ce76/spec/controllers/api/v1/tokens_controller_spec.rb#L89"></span>Useless assignment to variable - <code>user</code>.</td>
    </tr>
  </tbody>
</table>